src/benchmarklib/synthetic_table_generator.hpp

```diff
@@ -0,0 +1,139 @@
+    return c;
+  }
+
+  static ColumnDataDistribution make_skewed_normal_config(double skew_location = 0.0, double skew_scale = 1.0,
+                                                          double skew_shape = 0.0) {
+    ColumnDataDistribution c{};
+    c.skew_location = skew_location;
+    c.skew_scale = skew_scale;
+    c.skew_shape = skew_shape;
+    c.distribution_type = DataDistributionType::NormalSkewed;
+    return c;
+  }
+
+  DataDistributionType distribution_type = DataDistributionType::Uniform;
+
+  int num_different_values = 1'000;
+
+  double pareto_scale = 1.0;
+  double pareto_shape = 1.0;
+
+  double skew_location = 0.0;
+  double skew_scale = 1.0;
+  double skew_shape = 0.0;
+
+  double min_value = 0.0;
+  double max_value = 1.0;
+};
+
+class SyntheticTableGenerator {
+  // Note: numa_distribute_chunks=true only affects generated tables that use DictionaryCompression,
+  // otherwise the chunks are most likely all placed on a single node. This might change in the future.
+  // See the discussion here https://github.com/hyrise/hyrise/pull/402
+ public:
+  // Simple table generation, mainly for simple tests
+  std::shared_ptr<Table> generate_table(const size_t num_columns, const size_t num_rows, const ChunkOffset chunk_size,
+                                        const SegmentEncodingSpec segment_encoding_spec = {EncodingType::Unencoded});
+
+  std::shared_ptr<Table> generate_table(const std::vector<ColumnDataDistribution>& column_data_distributions,
+                                        const std::vector<DataType>& column_data_types, const size_t num_rows,
+                                        const ChunkOffset chunk_size,
+                                        const std::vector<SegmentEncodingSpec>& segment_encoding_specs,
+                                        const std::optional<std::vector<std::string>> column_names = std::nullopt,
+                                        const UseMvcc use_mvcc = UseMvcc::No,
+                                        const bool numa_distribute_chunks = false);
+
+  // Base function that generates the actual data
+  std::shared_ptr<Table> generate_table(const std::vector<ColumnDataDistribution>& column_data_distributions,
+                                        const std::vector<DataType>& column_data_types, const size_t num_rows,
+                                        const ChunkOffset chunk_size,
+                                        const std::optional<std::vector<std::string>> column_names = std::nullopt,
+                                        const UseMvcc use_mvcc = UseMvcc::No,
+                                        const bool numa_distribute_chunks = false);
+
+  /**
+    * Function to cast an integer to the requested type.
+    *   - in case of long, the integer is simply casted
+    *   - in case of floating types, the integer slighlty modified
+    *     and casted to ensure a matissa that is not fully zero'd
```

Comment:
There have been (actually are, but we mostly don't care) issues with comparing floats and using radix clustering (using the least significant bits for clustering where were all zero), which occurred from time to time. This is just adding more noice. I'll add a comment.

Reply:
`*   - in case of floating types, the integer slighlty modified and casted to ensure a matissa that is not fully
    *     zero'd. We have stumpled about operators that use radix clustering (e.g., some joins) and which did not show
    *     certain problems when all floating values had a mostly zero'ed matissa. The modification adds noice to make
    *     render the floating point values in some sense more realistic.`

Ok?

Reply:
*is slightly
*stumbled
*zeroed
*that
*mantissa
*noise
*s/render//

Please have your morning coffee before you continue typing ;)

Reply:
But does it make sense to you? You grammar nazi.

Reply:
Also added a bit more context:

```
* Function to create a typed value from an integer. The data generation creates integers with the requested
    * distribution and this function is used to create different types. The creation should guarantee that searching
    * for the integer value 100 within a column of the range (0,200) should return half of all tuples, and this also
    * for all other value types such as strings.
```

Reply:
Suggestion "should return half of all tuples, not only for ints but also for all other value types such as strings."